tests/modcc/test_printers.cpp

```diff
@@ -1,55 +1,93 @@
+#include <regex>
+#include <string>
+
 #include "test.hpp"
 
 #include "cprinter.hpp"
+#include "cudaprinter.hpp"
+#include "expression.hpp"
+#include "textbuffer.hpp"
+
+struct testcase {
+    const char* source;
+    const char* expected;
+};
+
+static std::string strip(std::string text) {
+    static std::regex rx(R"(\s)");
+    return std::regex_replace(text, rx, "");
+}
 
-using scope_type = Scope<Symbol>;
-using symbol_map = scope_type::symbol_map;
-using symbol_ptr = Scope<Symbol>::symbol_ptr;
-
-TEST(CPrinter, statement) {
-    std::vector<const char*> expressions =
-    {
-"y=x+3",
-"y=y^z",
-"y=exp(x/2 + 3)",
+TEST(scalar_printer, statement) {
+    std::vector<testcase> testcases = {
+        {"y=x+3",            "y=x+3"},
+        {"y=y^z",            "y=std::pow(y,z)"},
+        {"y=exp((x/2) + 3)", "y=exp(x/2+3)"},
+        {"z=a/b/c",          "z=a/b/c"},
+        {"z=a/(b/c)",        "z=a/(b/c)"},
+        {"z=(a*b)/c",        "z=a*b/c"},
+        {"z=a-(b+c)",        "z=a-(b+c)"},
+        {"z=(a>0)<(b>0)",    "z=a>0<(b>0)"}
```

Comment:
How about a test for `z=a- -2` to test the case of subtracting a negative value.